pkg/authorization/registry/subjectaccessreview/rest.go

```diff
@@ -33,18 +34,27 @@ func (r *REST) New() runtime.Object {
 func (r *REST) Create(ctx kapi.Context, obj runtime.Object) (runtime.Object, error) {
 	subjectAccessReview, ok := obj.(*authorizationapi.SubjectAccessReview)
 	if !ok {
-		return nil, kerrors.NewBadRequest(fmt.Sprintf("not a subjectAccessReview: %#v", obj))
+		return nil, kapierrors.NewBadRequest(fmt.Sprintf("not a subjectAccessReview: %#v", obj))
 	}
 	if err := kutilerrors.NewAggregate(authorizationvalidation.ValidateSubjectAccessReview(subjectAccessReview)); err != nil {
 		return nil, err
 	}
+	// if a namespace is present on the request, then the namespace on the SAR must be empty.  In such a case,
+	// the namespace on the SAR is overwritten.  This is to support backwards compatibility.  To have gotten here
+	// in this state, it means that the authorizer decided that a user could run an SAR against this namespace
+	if namespace := kapi.NamespaceValue(ctx); len(namespace) > 0 {
```

Comment:
ditto on comment and limiting the local check to when a namespace exists on the request